### PR TITLE
feat(java): Update to Java queries

### PR DESCRIPTION
**Pull request by GeekMasher**

### CODE REVIEW

File: java/lib/ghsl/CommandInjectionRuntimeExec.qll [hunk 1]
```diff
@@ -4,124 +4,111 @@ import semmle.code.java.dataflow.DataFlow
...
-        node.asExpr().getFile().isSourceFile() and
-        (
-            node.getType() instanceof PrimitiveType
-            or node.getType() instanceof BoxedType
-        )
-    }
-}
+  module ExecTaintFlow = TaintTracking::Global<ExecTaintConfiguration>;
 
+  import ExecTaintFlow::PathGraph
```
Comment (pwntester) [inline java/lib/ghsl/CommandInjectionRuntimeExec.qll:68]: I think this is not necessary. Only to bring in the edges predicate necessary for path-problems

File: java/lib/ghsl/CommandInjectionRuntimeExec.qll [hunk 2]
```diff
@@ -4,124 +4,111 @@ import semmle.code.java.dataflow.DataFlow
...
+  }
 
-abstract class Source extends DataFlow::Node {
-    Source() {
-        this = this
-    }
-}
+  module RuntimeExecFlow = TaintTracking::Global<RuntimeExecConfiguration>;
 
+  import RuntimeExecFlow::PathGraph
```
Comment (pwntester) [inline java/lib/ghsl/CommandInjectionRuntimeExec.qll:41]: I think this is not necessary. Only to bring in the edges predicate necessary for path-problems

File: java/lib/ghsl/CommandInjectionRuntimeExec.qll [hunk 3]
```diff
@@ -4,124 +4,111 @@ import semmle.code.java.dataflow.DataFlow
 private import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.dataflow.FlowSources
 
+abstract class Source extends DataFlow::Node {
+  Source() { this = this }
```
Comment (pwntester) [inline java/lib/ghsl/CommandInjectionRuntimeExec.qll:8]: Is this charpred needed?

Comment (pwntester) [reply java/lib/ghsl/CommandInjectionRuntimeExec.qll:8]: What classes extend it?

File: java/lib/qlpack.yml [hunk 4]
```diff
@@ -2,4 +2,5 @@ library: true
 name: githubsecuritylab/codeql-java-libs
 version: 0.0.1
 dependencies:
-  codeql/java-all: '*'
+  codeql/java-all: '1.0.0'
```
Comment (pwntester) [inline java/lib/qlpack.yml:5]: Should we use '^1.0.0' or better pin to an exact version?

File: java/src/qlpack.yml [hunk 5]
```diff
@@ -4,5 +4,5 @@ version: 0.0.5
 suites: suites
 defaultSuiteFile: suites/java.qls
 dependencies:
-  codeql/java-all: "*"
+  codeql/java-all: '1.0.0'
```
Comment (pwntester) [inline java/src/qlpack.yml:7]: Same question

File: java/test/qlpack.yml [hunk 6]
```diff
@@ -1,8 +1,8 @@
 name: githubsecurtylab/codeql-java-tests
 groups: [java, test]
 dependencies:
-    codeql/java-all: '*'
-    codeql/java-queries: '*'
+    codeql/java-all: '1.0.0'
+    codeql/java-queries: '1.0.0'
```
Comment (pwntester) [inline java/test/qlpack.yml:5]: Same question

File: java/lib/ghsl/CommandInjectionRuntimeExec.qll (hunk 3 above)
Comment (GeekMasher) [reply java/lib/ghsl/CommandInjectionRuntimeExec.qll:8]: I didn't add this and I honestly not sure how this impacts the query

**Pull request merged**